commands.py

```diff
@@ -96,32 +96,58 @@ def run(self, _: sublime.Edit) -> None:
 class CopilotAcceptSuggestionCommand(CopilotTextCommand):
     @_provide_session()
     def run(self, session: Session, edit: sublime.Edit) -> None:
-        completion = Completion(self.view)
-        if not completion.is_visible:
+        completion_manager = ViewCompletionManager(self.view)
+        if not completion_manager.is_visible:
             return
 
-        completion.hide()
+        completion_manager.hide()
+
+        completion = completion_manager.current_completion
+        if not completion:
+            return
 
         # Remove the current line and then insert full text.
         # We don't have to care whether it's an inline completion or not.
-        source_line_region = self.view.line(completion.region[1])
+        source_line_region = self.view.line(completion["positionSt"])
         self.view.erase(edit, source_line_region)
-        self.view.insert(edit, source_line_region.begin(), completion.text)
+        self.view.insert(edit, source_line_region.begin(), completion["text"])
+
+        # notify the current completion as accepted
+        self._record_telemetry(session, REQ_NOTIFY_ACCEPTED, {"uuid": completion["uuid"]})
 
-        # TODO: When a suggestion is accept, we need to send a REQ_NOTIFY_REJECTED
-        # request with all other completions which weren't accepted
-        self._record_telemetry(session, REQ_NOTIFY_ACCEPTED, {"uuid": completion.uuid})
+        # notify all other completions as rejected
+        other_uuids = [completion["uuid"] for completion in completion_manager.completions]
+        other_uuids.remove(completion["uuid"])
+        if other_uuids:
+            self._record_telemetry(session, REQ_NOTIFY_REJECTED, {"uuids": other_uuids})
 
 
 class CopilotRejectSuggestionCommand(CopilotTextCommand):
     @_provide_session()
     def run(self, session: Session, _: sublime.Edit) -> None:
-        completion = Completion(self.view)
-        completion.hide()
+        completion_manager = ViewCompletionManager(self.view)
+        completion_manager.hide()
+
+        # notify all completions as rejected
+        self._record_telemetry(
+            session,
+            REQ_NOTIFY_REJECTED,
+            {"uuids": [completion["uuid"] for completion in completion_manager.completions]},
+        )
+
 
-        # TODO: Currently we send the last shown completion UUID, however Copilot can
-        # suggest multiple UUID's. We need to return all UUID's which were not accepted
-        self._record_telemetry(session, REQ_NOTIFY_REJECTED, {"uuids": [completion.uuid]})
+class CopilotPreviousSuggestionCommand(CopilotTextCommand):
+    def run(self, _: sublime.Edit) -> None:
+        completion_manager = ViewCompletionManager(self.view)
+        completion_manager.choose_previous_completion()
+        completion_manager.show()
```

Comment:
Maybe we should call `show` directly in `choose_previous_completion`/`choose_next_completion`?

We could even rename `choose_next_completion` to just `next_completion` since `show` will be called under the hood.

So the command will be as simple as `ViewCompletionManager(self.view).previous_completion()`

Reply:
LGTM. Will do later.

Reply:
Done in [262207e](https://github.com/TheSecEng/LSP-copilot/pull/13/commits/262207e62bc78731d422214cc33f7114a3c3829c) though I use different method naming.